packages/backend/src/modules/tracked-txs/TrackedTxsIndexer.ts

```diff
@@ -38,88 +42,98 @@ export class LivenessIndexer extends ChildIndexer {
-      await this.livenessRepository.addMany(data, trx)
+    await this.configRepository.runInTransaction(async (trx) => {
+      for (const [type, updater] of Object.entries(this.updaters)) {
+        const filteredTxs = txs.filter((tx) => tx.use.type === type)
+        await updater.update(filteredTxs, trx)
+      }
 
-      await this.configurationRepository.setLastSyncedTimestamp(
-        configurations.map((c) => c.id),
-        adjustedTo,
-        trx,
+      const configIds = configurations.flatMap((config) =>
+        config.uses.map((use) => use.id),
       )
+      await this.configRepository.setLastSyncedTimestamp(configIds, syncTo, trx)
     })
 
     this.logger.info('Updated', {
       from,
-      adjustedTo: adjustedTo,
+      adjustedTo: syncTo,
       usedConfigurations: configurations.length,
-      fetchedDataPoints: data.length,
+      fetchedTxsCount: txs.length,
     })
-    return adjustedTo.toNumber()
+    return syncTo.toNumber()
   }
 
   async getConfiguration(
     from: number,
     to: number,
-  ): Promise<[LivenessConfigEntry[], UnixTime]> {
+  ): Promise<[TrackedTxConfigEntry[], UnixTime]> {
     const adjustedTo = adjustToForBigqueryCall(from, to)
 
-    const databaseEntries = await this.configurationRepository.getAll()
+    const databaseEntries = await this.configRepository.getAll()
 
-    const configurationsToSync = findConfigurationsToSync(
-      this.runtimeConfigurations,
+    const { configurationsToSync, syncTo } = findConfigurationsToSync(
+      this.configs,
       databaseEntries,
       new UnixTime(from),
       adjustedTo,
     )
 
-    return [configurationsToSync, adjustedTo]
+    return [configurationsToSync, syncTo]
   }
 
-  private async initialize() {
-    const databaseEntries = await this.configurationRepository.getAll()
-    const { toAdd, toRemove, toTrim } = diffLivenessConfigurations(
-      this.runtimeConfigurations,
+  private async initialize(): Promise<void> {
+    this.logger.info('Initializing...')
+    const databaseEntries = await this.configRepository.getAll()
+    const { toAdd, toRemove, toTrim } = diffTrackedTxConfigurations(
+      this.configs,
       databaseEntries,
     )
 
+    this.logger.info('Modifying tracked txs configs', {
+      added: toAdd.map((add) => add.id),
+      removed: toRemove,
+      trimmed: toTrim.map((trim) => trim.id),
+    })
+
     const safeHeight = getSafeHeight(databaseEntries, toAdd, this.minTimestamp)
 
     await this.stateRepository.runInTransaction(async (trx) => {
-      await this.configurationRepository.addMany(toAdd, trx)
-      // this will also delete records from "liveness" using CASCADE constraint
-      await this.configurationRepository.deleteMany(toRemove, trx)
+      await this.configRepository.addMany(toAdd, trx)
+      await this.configRepository.deleteMany(toRemove, trx)
       await this.trimConfigurations(toTrim, trx)
       await this.initializeIndexerState(safeHeight, trx)
```

Comment:
Maybe inside this function add a log that it was initialiezed for the first time